Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -297,6 +306,9 @@ protected override bool OnBackButtonPressed()
 
 		void InsertPageBefore(Page page, Page before)
 		{
+			if (page == null || before == null)
+				throw new ArgumentException($"Both {nameof(page)} and {nameof(before)} must be initialized.");
```

Comment:
I'm sure you heard about ArgumentNullException (). Or we can have a lengthy conversation about what "initialized" means in this context...